### PR TITLE
chore: convert react native project to typescript

### PR DESCRIPTION
**Pull request by jon-nona**
## Scope

- PT-38 - convert iso-ts-react-native to typescript

## Work Done

- convert project to typescript
- update eslint rules for typescript
- add jest watch typeahead for better test filtering
- update outdated dependencies
- update bundling
- change init ios command to use npm package pod-install.
- fix start reset command to reset metro bundler

## Steps to Test

- pull the branch
- install the dependencies and install your pods
- run the app on android and ios and make sure you get the following screenshots

## Screenshot
ScreenCapture at Sun Nov 15 21:58:25 SAST 2020

### CODE REVIEW

File: packages/mob-ui-react-native/index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { AppRegistry } from 'react-native'
-import App from './src/modules/App'
+import App from './src/modules/App/index'
```
Comment (neilrussell6) [inline packages/mob-ui-react-native/index.js:2]: why?

File: packages/mob-ui-react-native/package.json [hunk 2]
```diff
@@ -21,11 +21,11 @@
   "scripts": {
     "serve:android": "react-native run-android",
     "serve:ios": "react-native run-ios",
-    "init:ios": "cd ios && pod install",
+    "init:ios": "npx pod-install",
```
Comment (neilrussell6) [inline packages/mob-ui-react-native/package.json:24]: shouldn't need npx in npm script context if pod-install is a dep (can you add as dep)?

File: packages/mob-ui-react-native/src/modules/App/App.component.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { SafeAreaView, Text } from 'react-native'
 
 import CommonStyles from '../../common/styles'
 
-const App = () => (
+const App: React.FunctionComponent<Record<string, unknown>> = () => (
```
Comment (neilrussell6) [inline packages/mob-ui-react-native/src/modules/App/App.component.tsx:6]: in React we are using:
```
import React, { FC } from 'react'

const App: FC = () => (
```
is there nothing similarly simple in RN?

**Review by neilrussell6 — CHANGES_REQUESTED** [review]

File: packages/mob-ui-react-native/src/modules/App/App.component.tsx (hunk 3 above)
Comment (jon-nona) [reply packages/mob-ui-react-native/src/modules/App/App.component.tsx:6]: @neilrussell6 FC is an alias of FunctionComponent? I just prefer the latter as it's more explicit, and I guess as far as typing the empty props interface goes, I can remove that if you want?

**Review by neilrussell6 — APPROVED** [review]

**Pull request merged**